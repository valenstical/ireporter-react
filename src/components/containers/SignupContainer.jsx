import React, { Component } from 'react';
import { connect } from 'react-redux';
import formDataJSON from 'formdata-json';
import PropTypes from 'prop-types';
import { createStructuredSelector } from 'reselect';
import { Redirect } from 'react-router-dom';
import { getUser } from '../../selectors';
import { scrollTop } from '../../utils/helpers';
import Signup from '../views/Signup';
import authenticate from '../../actions/authAction';

/**
 * Container component for the Singup view
 * @export
 * @class SignupContainer
 * @extends {Component}
 */
class SignupContainer extends Component {
  handleSignup = async (event) => {
    event.preventDefault();
    const payload = formDataJSON(new FormData(event.target));
    payload.route = 'auth/signup';
    payload.method = 'post';
    const { signup: processSignup } = this.props;
    await processSignup(payload);
    scrollTop();
  }

  render() {
    const { user } = this.props;
    if (user.success || user.isLoggedIn) return <Redirect to="/dashboard" />;
    return (
      <Signup
      handleSubmit={this.handleSignup}
      user={user}
      />
    );
  }
}
SignupContainer.propTypes = {
  user: PropTypes.object.isRequired,
  signup: PropTypes.func.isRequired
};

const mapStateToProps = createStructuredSelector(
  {
    user: getUser
  }
);

const mapDispatchToProps = dispatch => ({
  signup: payload => dispatch(authenticate(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(SignupContainer);
