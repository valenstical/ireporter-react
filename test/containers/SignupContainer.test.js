import React from 'react';
import setup from '../setup';
import SignupContainer from '../../src/components/containers/SignupContainer';


describe('<SignupContainer>', () => {
  it('should render without crashing', () => {
    const wrapper = setup(<SignupContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
