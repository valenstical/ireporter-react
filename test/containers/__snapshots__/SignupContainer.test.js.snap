// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupContainer> should render without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(SignupContainer)>
        <SignupContainer
          signup={[Function]}
          user={
            Object {
              "isBusy": false,
              "message": Array [],
              "success": false,
            }
          }
        >
          <Signup
            handleSubmit={[Function]}
            user={
              Object {
                "isBusy": false,
                "message": Array [],
                "success": false,
              }
            }
          >
            <section
              className="login-wrapper bg-grey top-space"
              data-pg-name="How it Works"
            >
              <form
                className="wrapper"
                data-pg-name="Wrapper"
                noValidate={true}
                onSubmit={[Function]}
              >
                <div
                  className="card login-form login-form-lg"
                >
                  <h1
                    className="text-brand"
                  >
                    WELCOME TO iREPORTER
                  </h1>
                  <p>
                    Let's help you quickly report cases of corruption and issues that need government attention to the appropriate authorities.
                    <span>
                       Register by filling out the form below to get started.
                    </span>
                  </p>
                  <small
                    className="form-instructions"
                  >
                    * All fields are required
                  </small>
                  <div
                    className="form-wrapper"
                  >
                    <label
                      htmlFor="firstname"
                    >
                      First Name 
                    </label>
                    <input
                      className="form-element"
                      id="firstname"
                      name="firstname"
                      required={true}
                      type="text"
                    />
                  </div>
                  <div
                    className="form-wrapper"
                  >
                    <label
                      htmlFor="lastname"
                    >
                      Last Name 
                    </label>
                    <input
                      className="form-element"
                      id="lastname"
                      name="lastname"
                      required={true}
                      type="text"
                    />
                  </div>
                  <div
                    className="form-wrapper"
                  >
                    <label
                      htmlFor="othernames"
                    >
                      Other Names
                      <small>
                        (optional)
                      </small>
                    </label>
                    <input
                      className="form-element"
                      id="othernames"
                      name="othernames"
                      required={true}
                      type="text"
                    />
                  </div>
                  <div
                    className="form-wrapper"
                  >
                    <label
                      htmlFor="phone"
                    >
                      Phone Number 
                    </label>
                    <input
                      className="form-element"
                      id="phone"
                      name="phoneNumber"
                      required={true}
                      type="tel"
                    />
                  </div>
                  <div
                    className="form-wrapper"
                  >
                    <label
                      htmlFor="email"
                    >
                      Email Address 
                    </label>
                    <input
                      className="form-element"
                      id="email"
                      name="email"
                      required={true}
                      type="email"
                    />
                  </div>
                </div>
                <div
                  className="card login-form login-form-lg"
                >
                  <div
                    className="form-wrapper"
                  >
                    <label
                      htmlFor="username"
                    >
                      Choose a Username 
                    </label>
                    <input
                      className="form-element"
                      id="username"
                      name="username"
                      required={true}
                      type="text"
                    />
                  </div>
                  <div
                    className="form-wrapper"
                  >
                    <label
                      htmlFor="password"
                    >
                      Create a Password
                    </label>
                    <input
                      autoComplete="off"
                      className="form-element"
                      id="password"
                      name="password"
                      required={true}
                      type="password"
                    />
                  </div>
                  <div
                    className="checkbox"
                  >
                    <label>
                      <input
                        id="termsCheckedBox"
                        required={true}
                        type="checkbox"
                      />
                      I agree to the
                      <Link
                        className="forgot-password"
                        replace={false}
                        to="#"
                      >
                        <a
                          className="forgot-password"
                          href="/"
                          onClick={[Function]}
                        >
                          Terms and Conditions
                        </a>
                      </Link>
                    </label>
                  </div>
                  <button
                    className="btn-brand btn-submit"
                    type="submit"
                  >
                    <span>
                      Register Now
                    </span>
                  </button>
                </div>
                <p
                  className="login-register-link login-form-lg"
                >
                  Already registered?
                   
                  <Link
                    replace={false}
                    to="/login"
                  >
                    <a
                      href="/login"
                      onClick={[Function]}
                    >
                      Sign in
                    </a>
                  </Link>
                </p>
              </form>
            </section>
          </Signup>
        </SignupContainer>
      </Connect(SignupContainer)>
    </Router>
  </BrowserRouter>
</Provider>
`;
